src/platform/cmx/boot_comp.c

```diff
@@ -0,0 +1,424 @@
+	if (!uvm4k0pgd_cap || !CAP_TYPECHK(uvm4k0pgd_cap, CAP_PGTBL)) assert(0);
+	uvm4k0pgtbl = (pgtbl_t)&(uvm4k0pgd_cap->pgtbl);
+
+	uvm4k0pgtbl->data.pgt[0]=(COS_CMX_SECOND_PAGE+0x0000)|CMX_PGTBL_COSFRAME;
+	uvm4k0pgtbl->data.pgt[1]=(COS_CMX_SECOND_PAGE+0x1000)|CMX_PGTBL_COSFRAME;
+	uvm4k0pgtbl->data.pgt[2]=(COS_CMX_SECOND_PAGE+0x2000)|CMX_PGTBL_COSFRAME;
+	uvm4k0pgtbl->data.pgt[3]=(COS_CMX_SECOND_PAGE+0x3000)|CMX_PGTBL_COSFRAME;
+	uvm4k0pgtbl->data.pgt[4]=(COS_CMX_SECOND_PAGE+0x4000)|CMX_PGTBL_COSFRAME;
+	uvm4k0pgtbl->data.pgt[5]=(COS_CMX_SECOND_PAGE+0x5000)|CMX_PGTBL_COSFRAME;
+	uvm4k0pgtbl->data.pgt[6]=(COS_CMX_SECOND_PAGE+0x6000)|CMX_PGTBL_COSFRAME;
+	uvm4k0pgtbl->data.pgt[7]=(COS_CMX_SECOND_PAGE+0x7000)|CMX_PGTBL_COSFRAME;
+
+	uvm4k1pgd_cap = (struct cap_pgtbl*)captbl_lkup(ct, uvm4k1pgdcap);
+	if (!uvm4k1pgd_cap || !CAP_TYPECHK(uvm4k1pgd_cap, CAP_PGTBL)) assert(0);
+	uvm4k1pgtbl = (pgtbl_t)&(uvm4k1pgd_cap->pgtbl);
+
+	uvm4k1pgtbl->data.pgt[0]=(COS_CMX_SECOND_PAGE+0x8000)|CMX_PGTBL_COSFRAME;
+	uvm4k1pgtbl->data.pgt[1]=(COS_CMX_SECOND_PAGE+0x9000)|CMX_PGTBL_COSFRAME;
+	uvm4k1pgtbl->data.pgt[2]=(COS_CMX_SECOND_PAGE+0xA000)|CMX_PGTBL_COSFRAME;
+	uvm4k1pgtbl->data.pgt[3]=(COS_CMX_SECOND_PAGE+0xB000)|CMX_PGTBL_COSFRAME;
+	uvm4k1pgtbl->data.pgt[4]=(COS_CMX_SECOND_PAGE+0xC000)|CMX_PGTBL_COSFRAME;
+	uvm4k1pgtbl->data.pgt[5]=(COS_CMX_SECOND_PAGE+0xD000)|CMX_PGTBL_COSFRAME;
+	uvm4k1pgtbl->data.pgt[6]=(COS_CMX_SECOND_PAGE+0xE000)|CMX_PGTBL_COSFRAME;
+	uvm4k1pgtbl->data.pgt[7]=(COS_CMX_SECOND_PAGE+0xF000)|CMX_PGTBL_COSFRAME;
+
+	/* Setup the booter component's page table then */
+	pgd_cap = (struct cap_pgtbl*)captbl_lkup(ct, pgdcap);
+	if (!pgd_cap || !CAP_TYPECHK(pgd_cap, CAP_PGTBL)) assert(0);
+	pgtbl = (pgtbl_t)&(pgd_cap->pgtbl);
+
+	/* Add kernel data regions - they shall never be retyped into anything else */
+	pgtbl->data.pgt[0]=0;
+	pgtbl->data.pgt[1]=0;
+	pgtbl->data.pgt[2]=0;
+	pgtbl->data.pgt[3]=COS_CMX_FIRST_PAGE|CMX_PGTBL_USER_DEF;
+	/* pgtbl->data.pgt[4]=0; something cons'ed into here already */
+	/* This part is erased from our sight already
+	pgtbl->data.pgt[5]=COS_CMX_THIRD_PAGE|PGTBL_COSFRAME;
+	pgtbl->data.pgt[6]=COS_CMX_FOURTH_PAGE|PGTBL_COSFRAME;
+	pgtbl->data.pgt[7]=COS_CMX_FIFTH_PAGE|PGTBL_COSFRAME; */
+
+	/* Retype the first component's range to user memory */
+	for(unsigned long i=COS_CMX_FIRST_PAGE;i<COS_CMX_SECOND_PAGE;i+=RETYPE_MEM_SIZE)
+		assert(retypetbl_retype2user((void*)i)==0);
+
+	/* Manually construct the initial MPU settings. After this, all the settings must be constructed by the update function.
+	 * here, we only grant access to the first MPU regions. All other code regions, by default, are enabled; and the background
+	 * region is also enabled. These are done by default */
+	chal_mpu_update(pgtbl, 1);
+	return 0;
+}
+
+/* PRY:we have two chronos now - this shall be fixed later */
+static void
+kern_boot_frt(struct captbl *ct, void *thd_mem, void *tcap_mem)
+{
```

Comment:
If `frt` means `FreeRTOS`, I'd prefer to use `frtos`.
Also, if it is, perhaps move it to a separate `frtos_boot.c` or something. ;)